cpp/src/gandiva/regex_util.h

```diff
@@ -29,7 +29,7 @@ namespace gandiva {
 /// \brief Utility class for converting sql patterns to pcre patterns.
 class GANDIVA_EXPORT RegexUtil {
  public:
-  // Convert an sql pattern to a pcre pattern
+  // Convert an sql pattern to a pcre pattern for use with PartialMatch
```

Comment:
Previously, `like` was using `RE2::FullMatch`. However, `regexp_matches` needs to use `RE2::PartialMatch`, so I've modified this method to make partial match statements. I have added a new set of tests to validate it.

I didn't see any use of this utility method outside of the `like` implementation, so I thought this change would be okay.